src/cartesian/Area.tsx

```diff
@@ -661,7 +661,7 @@ class AreaWithState extends PureComponent<InternalProps> {
         />
         {this.props.isRange && Array.isArray(baseLine) && (
           <ActivePoints
-            points={baseLine}
+            points={baseLine.map((pt, idx) => ({ ...pt, payload: points[idx]?.payload }))}
```

Comment:
why do we need to add `payload` here? Looks like we didn't have to previously

Reply:
I too prefer to have this done in the selector, because:

1. It's easier to test the selector
2. Keeps the React component (which is effectively a view layer) small and focused
3. We will eventually want to publish the selector as a hook, and we will want external consumers to have access to same information as internal

I almost wrote "consistent" but in Recharts everything is inconsistent with everything so I won't use that as an argument